modules/debts_api/lib/debts_api/v0/fsr_form_transform/enhanced_expense_calculator.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+require 'debts_api/v0/fsr_form_transform/expense_calculator'
+
+module DebtsApi
+  module V0
+    module FsrFormTransform
+      class EnhancedExpenceCalculator
+        RENT = 'Rent'
+        MORTGAGE_PAYMENT = 'Mortgage payment'
+        FOOD = 'Food'
+
+        def initialize(form)
+          @form = form
+          @enhanced = form['view:enhancedFinancialStatusReport'] || false
+          @expense_records = @form.dig('expenses', 'expenseRecords') || []
+          @old_rent_mortgage_attr = @form.dig('expenses', 'rentOrMortgage')
+          @old_food_attr = @form.dig('expenses', 'food')
+          @credit_card_bills = @form.dig('expenses', 'creditCardBills') || []
+          @other_expenses = @form['otherExpenses'].deep_dup || []
+          @installment_contracts = @form['installmentContracts']
+          @utility_records = @form['utilityRecords']
+
+          @filtered_expenses = [].concat(
+            exclude_by(@other_expenses, [FOOD]),
+            exclude_by(@expense_records, [RENT, MORTGAGE_PAYMENT])
+          )
+          @all_expenses ||= get_all_expenses
+        end
+
+        def get_monthly_expenses
+          utilities = get_utilities
+          installments = safe_sum(@installment_contracts.pluck('amountDueMonthly'))
+          other_exp = safe_sum(@other_expenses.pluck('amount'))
+          calculated_exp_records = safe_sum(@expense_records.pluck('amount'))
+          food = safe_number(@old_food_attr)
+          rent_or_mortgage_sum = safe_number(@old_rent_mortgage_attr)
+          credit_card_bills = safe_sum(@credit_card_bills.pluck('amountDueMonthly'))
+
+          sum = utilities +
+                installments +
+                other_exp +
+                calculated_exp_records +
+                food +
+                rent_or_mortgage_sum +
+                credit_card_bills
+
+          sum.round(2)
+        end
+
+        def get_all_expenses
```

Comment:
Can we avoid using `get_` here? Accessors/writers in ruby aren't typically prefixed with get_/set_.